molecule/multimachine/cleanup.yml

```diff
@@ -30,7 +30,7 @@
   hosts: localhost
   gather_facts: false
   vars:
-    broker_name: molecule-{{ lookup('env', 'BUILD_NUMBER') }}
+    broker_name: molecule-{{ lookup('env', 'BRANCH_NAME') }}-{{ lookup('env', 'BUILD_NUMBER') }}
```

Comment:
I remember that not using branch_name was intentional for the usual too long limitation in the name

we could enforce via precommit or truncate branch_name to avoid that